### DISCUSSION TITLE
How to merge duplicate data

### DISCUSSION THREAD

**Discussion by libook** [containrrr/watchtower#1815] [Q&A] [answered]
These metrics always return the result during the last scan:
- watchtower_containers_failed
- watchtower_containers_scanned
- watchtower_containers_updated

That is, if watchtower scans once a day and prometheus collects data every hour, then the vast majority of data collected during the day is from the same scan.
There will be a lot of duplicates in the final statistics presented, but I'm not sure what can be used to accurately merge the duplicates.

Is there any PromQL syntax or Grafana options that could merger these data?

Or is it possible for watchtower to work around this by adding attributes to the metrics (e.g. adding the last scan time to each metric)?

**Comment by piksel** [comment]
Hmm. I'm not sure how you are using the metrics, but those are gauges, meaning "the current status of...". Like, if you had a gauge of the "currently logged in users" and took a sample of its value, you couldn't really merge it with another sample. It's just a metric of what that metrics were showing at the time of measure.

You could of course transform this into something else, but that depends on how you want to visualize the data.

If you want to present them in a table, you could just use something like `delta(watchtower_containers_updated[24h])` to show the differences (which would be 0 between runs), or lower the prometheus collection rate.

Another way would be to group them by `watchtower_scan_count` which would only change once per day.

**Comment by libook** [comment, marked as answer]
I ended up taking one point per day by doing the following:

Query Options -> Min interval=1d

Query:
```
last_over_time(watchtower_containers_scanned[1d])
```
Options -> Min step=1d

![Edit-panel-Watchtower-仪表板-Grafana](https://github.com/containrrr/watchtower/assets/3395610/e08b05be-87b1-4969-8e78-c4d6e1e499b2)
